js/createRandomShowTimes.js

```diff
@@ -0,0 +1,117 @@
+
+            // tempTime is set to be altered for each day so tempDate dosent get alterd each day (note to self: cloud just reset the tempDate at the end of loop)
+            let tempTime = new Date(tempDate);
+            // sets the day to be able to controll start time for the first show, which varies from weekends 
+            let day = tempDate.getDay();
+            
+            // sundays open at 10:30, saturdays 12:30, all other days at 13:30
+            if(day == 0) {
+                tempTime.setHours(10, 30);
+            } else if (day == 6) {
+                tempTime.setHours(12, 30);
+            } else {
+                tempTime.setHours(13, 30);
+            }
+
+            showTimes[i].dates[j].showtimes.push({"time":`${tempTime.getHours()}:${tempTime.getMinutes()}`, "movieIndex":Math.floor(Math.random() * database.length)});
+            
+            // Goes through and sets the showtimes for each day, randomes a movie from the database
+            for(let k=0;tempTime.getHours() < 23;k++) {
+                
+                let movieIndex = showTimes[i].dates[j].showtimes[k].movieIndex;
+                let movieLength = database[movieIndex].length; // could just write database[showTimes[i].dates[j].showtimes[k].movieIndex].length , but it gets hard to read.
+                
+                // A test date to check to see if it goes over closing time or not before setting the real time
+                let temp2Time = new Date(tempTime);
+
+                // If movielength goes over 23:00 (closing time) then break the shit.
+                addMovieLengthToTime(movieLength, temp2Time)
+                if((temp2Time.getHours() < 23) && (temp2Time.getHours() > 10)){
+                    addMovieLengthToTime(movieLength, tempTime);
+                } else {
+                    break;
+                }
+                
+                // If cleanuptime goes over 23:00 (closing time) then break the shit.
+                nextShowTime(temp2Time);
+                if(temp2Time.getHours() < 23 && (temp2Time.getHours() > 10)){
+                    nextShowTime(tempTime);
+                } else{
+                    break;
+                }
+
+                // Just to make it prittier, other wise time will get stamped as 17:0 insted of 17:00
+                if (tempTime.getMinutes() == 0) {
+                    showTimes[i].dates[j].showtimes.push({"time":`${tempTime.getHours()}:00`, "movieIndex":Math.floor(Math.random() * database.length)});    
+                } else {
+                    showTimes[i].dates[j].showtimes.push({"time":`${tempTime.getHours()}:${tempTime.getMinutes()}`, "movieIndex":Math.floor(Math.random() * database.length)});
+                }
+
+            }
+            // Go to next day
+            tempDate.setDate(tempDate.getDate() + 1);
+        }
+        
+    } 
+
+    //console.log(showTimes);
+    console.log(JSON.stringify(showTimes));
```

Comment:
On the one hand, we don't need console.log in code on github. On the other hand, this file isn't meant for production, but to generate a new file if needed. So - in this particular case it doesn't matter, but something to keep in mind for the future! 😁

Reply:
Indeed you are right, "fetch("static/database.json")" works. When I tried to do something similar I got lots of errors, but this works, thanks!

Yes, the console.log was just left in out of lazyness to not need to write it again if we need to recreate the database. As you say the createRandomShowTimes.js is not needed other then in development. But I thought it was good to have a backup on the repo.